From f4135340535580fef5da41dc26456e0ba55716c8 Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Tue, 5 Feb 2019 20:44:10 +0530
Subject: [PATCH 126/255] octeontx2-af: Use the right wake_up API variant in
 CGX command interface

commit a684e233a527ca61ca2f3c032ce84ace4928f1ae from
git@git.assembla.com:cavium/WindRiver.linux.git

We use wait_event_timeout API for process waiting on command completion,
hence use wake_up API for waking up instead of wake_up_interruptible.

This improves the command completion waiting time.

Change-Id: I8a3ddebf8ed7e56183d6bcdbefadfb6e2e6a1a76
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 0392bdc5cd4a..8d75a66459ff 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -582,7 +582,7 @@ static irqreturn_t cgx_fwi_event_handler(int irq, void *data)
 
 		/* Release thread waiting for completion  */
 		lmac->cmd_pend = false;
-		wake_up_interruptible(&lmac->wq_cmd_cmplt);
+		wake_up(&lmac->wq_cmd_cmplt);
 		break;
 	case CGX_EVT_ASYNC:
 		if (cgx_event_is_linkevent(event))
-- 
2.17.1

